### PR TITLE
[core][autoscaler] Remove usage of grpcio from autoscaler SDK

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
Since we should not be using grpcio python (aligning with this https://github.com/ray-project/ray/issues/35472)

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/autoscaler/v2/tests/test_sdk.py [hunk 1]
```diff
@@ -6,15 +6,23 @@
...
-    _autoscaler_state_service_stub,
-    request_cluster_resources,
-)
+from ray.autoscaler.v2.sdk import request_cluster_resources
 from ray.autoscaler.v2.tests.util import get_cluster_resource_state
+from ray.core.generated.experimental import autoscaler_pb2_grpc
 from ray.core.generated.experimental.autoscaler_pb2 import GetClusterResourceStateReply
 
 
+def _autoscaler_state_service_stub():
```
Comment (rickyyx) [inline python/ray/autoscaler/v2/tests/test_sdk.py:17]: are we good with keeping grpcio in the test?

**Comment by rickyyx** [comment]
The test is at https://github.com/ray-project/ray/blob/374eb2c9e36b2309403125c3954f31a02d1f730e/python/ray/autoscaler/v2/tests/test_sdk.py#L49

File: python/ray/autoscaler/v2/sdk.py [hunk 2]
```diff
@@ -34,18 +32,7 @@ def request_cluster_resources(to_request: List[dict], timeout: int = 10):
         timeout: Timeout in seconds for the request to be timeout
 
     """
+    if not ray.is_initialized():
```
Comment (scv119) [inline python/ray/autoscaler/v2/sdk.py:35]: hmm why is this necessary?

Comment (rickyyx) [reply python/ray/autoscaler/v2/sdk.py:35]: so we can get the GCSClient that's initialized in the worker

Comment (rickyyx) [reply python/ray/autoscaler/v2/sdk.py:35]: hmm actually i think the caller already checks it.

**Review by scv119 — APPROVED** [review]
